backend/src/main/java/com/urlshortner/backend/controller/Controller.java

```diff
@@ -1,15 +1,76 @@
 package com.urlshortner.backend.controller;
 
+import com.urlshortner.backend.entity.UrlMapping;
+import com.urlshortner.backend.service.UrlMappingService;
 import lombok.extern.slf4j.Slf4j;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
 
 @RestController
 @Slf4j
 public class Controller {
     @GetMapping("/ping")
     public String getHello(){
-        log.info("hello");
+       // log.info("hello");
         return "hello";
     }
+
+    @Autowired
+    private UrlMappingService service;
+
+    private static final Logger logger = LoggerFactory.getLogger(Controller.class);
+
+
+    @PostMapping("/shorten")
+    @ResponseBody
+    public String shortenUrl(@RequestBody String originalUrl) {
+        logger.info("Received request to shorten URL: {}", originalUrl);
+
+        // Extract the path part of the URL (not including the domain)
+        String path = extractPath(originalUrl);
+
+        // Generate a short key based on the extracted path
+        String shortKey = generateShortKey(path);
```

Comment:
There can duplicate short key value . So, there is a chance of getting 2 Original Urls for the same short key.